src/Middleware/JsonSchema.php

```diff
@@ -30,31 +29,15 @@ public function __construct(array $schemas)
      *
      * @return ResponseInterface
      */
-    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, callable $next)
+    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, callable $next = null)
     {
         $schema = $this->getSchema($request);
 
         if (is_object($schema)) {
-            $value = $request->getParsedBody();
-            if (!is_object($value)) {
-                return $this->invalidateResponse(
-                    $response,
-                    sprintf('Parsed body must be an object. Type %s is invalid.', gettype($value))
-                );
-            }
-
-            $validator = new Validator();
-            $validator->check($value, $schema);
-
-            if (!$validator->isValid()) {
-                return $this->invalidateResponse(
-                    $response,
-                    'Unprocessable Entity',
-                    [
-                        'Content-Type' => 'application/json',
-                    ],
-                    json_encode($validator->getErrors(), JSON_UNESCAPED_SLASHES)
-                );
+            $validator = new JsonValidator($schema);
+            $response = $validator($request, $response);
```

Comment:
If this middleware is the same than `JsonValidator` but handle multiple schemes, once the scheme has been choosen, why not simply execute the following?

```php
return $validator($request, $response, $next);
```